File: datalib/src/main/java/nl/acidcats/tumblrlikes/core/usecases/photos/GetFilteredPhotoUseCase.java
package nl.acidcats.tumblrlikes.core.usecases.photos;

import nl.acidcats.tumblrlikes.core.models.Photo;
import rx.Observable;

/**
 * Created on 19/10/2018.
 */
public interface GetFilteredPhotoUseCase {
    Observable<Photo> getNextFilteredPhoto();
}

File: datalib/src/main/kotlin/nl/acidcats/tumblrlikes/core/usecases/photos/GetFilteredPhotoUseCase.kt
package nl.acidcats.tumblrlikes.core.usecases.photos

import nl.acidcats.tumblrlikes.core.models.Photo
import nl.acidcats.tumblrlikes.core.viewmodels.PhotoViewModel
import rx.Observable

/**
 * Created on 19/10/2018.
 */
interface GetFilteredPhotoUseCase {
    fun getNextFilteredPhoto(): Observable<PhotoViewModel>
}
